websites/M/MangaKatana/presence.ts

```diff
@@ -0,0 +1,184 @@
+
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
+
+	function getSearchState() {
+		if (!search) return title;
+		const genres = search?.match(/include=([A-Za-z_-\w]+)+/);
+		if (!genres) return;
+		return `${strings.genre}: ${genres[1]
+			?.split("_")
+			?.map((genre: string) => genre.charAt(0).toUpperCase() + genre.slice(1))
+			?.join(", ")}`;
+	}
+
+	switch (path[1]) {
+		case "":
+			presenceData.details = strings.viewHome;
+			break;
+
+		case "manga":
+			presenceData.details = privacy
+				? strings.searchSomething
+				: strings.searchFor;
+			presenceData.smallImageKey = Assets.Search;
+			presenceData.smallImageText = strings.search;
+			presenceData.state = getSearchState();
+
+			if (path[2]) {
+				presenceData.details = privacy
+					? `${strings.viewPage} ${strings.manga}`
+					: strings.readingAbout;
+				presenceData.state = textContent("#single_book .info .heading");
+				presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+					"#single_book .cover img"
+				)?.src;
+				presenceData.smallImageKey = Assets.Viewing;
+				presenceData.smallImageText = strings.viewing;
+				presenceData.buttons = [
+					{
+						label: strings.buttonViewPage,
+						url: href,
+					},
+				];
+			}
+
+			if (path[3]) {
+				presenceData.details = `${strings.reading} ${
+					privacy
+						? strings.manga
+						: textContent("ol.uk-breadcrumb li:nth-child(2) span")
+				}`;
+				presenceData.state = textContent(
+					"ol.uk-breadcrumb li:nth-child(3) span"
+				);
+
+				if (pages) {
+					for (const page of document.querySelectorAll("#imgs img"))
+						observer.observe(page);
+					presenceData.largeImageKey = localStorage.getItem("currentMangaPage");
+				} else presenceData.largeImageKey = Assets.Cover;
+
+				presenceData.smallImageKey = Assets.Reading;
+				presenceData.smallImageText = strings.reading;
+				presenceData.buttons = [
+					{
+						label: strings.buttonViewPage,
+						url: href,
+					},
+				];
+			}
+			break;
+
+		case "genres":
+		case "latest":
+		case "new-manga":
+		case "genre":
+		case "author":
+			presenceData.details = privacy
+				? strings.searchSomething
+				: strings.searchFor;
+			presenceData.smallImageKey = Assets.Search;
+			presenceData.smallImageText = strings.search;
+			presenceData.state = getSearchState();
+			break;
+
+		case "settings":
+		case "bookmarks":
+			presenceData.details = strings.editing;
+			presenceData.state = title;
+			presenceData.smallImageKey = Assets.Cover;
+			break;
+	}
+
+	if (searchInput) {
+		presenceData.details = privacy
+			? strings.searchSomething
+			: strings.searchFor;
+		presenceData.smallImageKey = Assets.Search;
+		presenceData.smallImageText = strings.search;
+		presenceData.state = searchInput;
+	}
+
+	if (!cover || privacy) presenceData.largeImageKey = Assets.Cover;
+	if (!buttons || privacy) delete presenceData.buttons;
```

Comment:
```suggestion
	if ((!buttons || privacy) && presenceData.buttons) delete presenceData.buttons;
```
Adding checks if they are defined before deleting them

Reply:
I'm not sure if these are necessary. delete won't throw an error if deleting somethat that is not defined, just return false.

Reply:
They aren't necessary at all, but were recommended to me by @DarkVillager. Which made me recommend it here
https://github.com/PreMiD/Presences/pull/7346#discussion_r1189580219

Reply:
> I'm not sure if these are necessary. delete won't throw an error if deleting somethat that is not defined, just return false.

Agreed but it'd be better imo.